publishing bot bug

When the robot tries to cherry-pick https://github.com/kubernetes/apiserver/commit/e701e4054475458134b6b25aae0ca96298704abd to k8s.io/apiserver master branch, the robot fails with:
```
working 628b5dd6c2e8df84d9ccfba5ebfc6ec00ff53b7b
The previous cherry-pick is now empty, possibly due to conflict resolution.
If you wish to commit it anyway, use:

    git commit --allow-empty

Otherwise, please use 'git reset'
On branch master
Your branch is up-to-date with 'origin/master'.
You are currently cherry-picking commit 628b5dd.
```

Yet another problem in client-go release-2.0
```
--- FAIL: TestIsConfigTransportTLS (0.00s)
	config_test.go:86: expected true for &rest.Config{Host:"1.2.3.4:567", APIPath:"", Prefix:"", ContentConfig:rest.ContentConfig{AcceptContentTypes:"", ContentType:"", GroupVersion:(*unversioned.GroupVersion)(nil), NegotiatedSerializer:runtime.NegotiatedSerializer(nil)}, Username:"", Password:"", BearerToken:"", Impersonate:"", AuthProvider:(*api.AuthProviderConfig)(nil), AuthConfigPersister:rest.AuthProviderConfigPersister(nil), TLSClientConfig:rest.TLSClientConfig{CertFile:"", KeyFile:"", CAFile:"", CertData:[]uint8(nil), KeyData:[]uint8(nil), CAData:[]uint8(nil)}, Insecure:true, UserAgent:"rest.test/v1.5.3 (linux/amd64) kubernetes/$Format", Transport:http.RoundTripper(nil), WrapTransport:(func(http.RoundTripper) http.RoundTripper)(nil), QPS:0, Burst:0, RateLimiter:flowcontrol.RateLimiter(nil), Timeout:0}
--- FAIL: TestValidatesHostParameter (0.00s)
	url_utils_test.go:51: unexpected error parse 127.0.0.1:8080: first path segment in URL cannot contain colon
FAIL
FAIL	k8s.io/client-go/rest	0.998s
```

--- a/mungegithub/mungers/publish_scripts/util.sh
+++ b/mungegithub/mungers/publish_scripts/util.sh
@@ -97,8 +97,8 @@ sync_repo() {
         if [[ -z "${commitSHA}" ]]; then
             continue
         fi
-    	echo "working ${commitSHA}"
-    	git -c user.name="Kubernetes Publisher" -c user.email="k8s-publish-robot@users.noreply.github.com" cherry-pick ${commitSHA}
+        echo "working ${commitSHA}"
+        git -c user.name="Kubernetes Publisher" -c user.email="k8s-publish-robot@users.noreply.github.com" cherry-pick --keep-redundant-commits ${commitSHA}
     done <<< "${commits}"
     
     # track the k8s.io/kubernetes commit SHA so we can always determine which level of kube this repo matches

